src/components/shared/CopyableAddressV2/CopyableAddressV2.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useCallback } from 'react';
+import clsx from 'clsx';
+import { splitAddress } from '~utils/strings';
+
+import { Address } from '~types';
+
+import MaskedAddress from '~shared/MaskedAddress';
+import useClipboardCopy from '~hooks/useClipboardCopy';
+import Icon from '~shared/Icon';
+// Note that we use CSS here because MaskedAddress css file will override font-normal by tailwind due to injection order
+import styles from './CopyableAddressV2.css';
+
+interface Props {
+  /** Address to display */
+  children: Address;
+  /** Indicates that the full address should be shown instead of an abbreviated one */
+  full?: boolean;
+  /** In some occasions we want to show the button to copy only */
+  hideAddress?: boolean;
+}
+
+const displayName = 'CopyableAddressV2';
+
+const CopyableAddressV2 = ({
+  children: address,
+  full,
+  hideAddress = false,
+}: Props) => {
+  const { isCopied, handleClipboardCopy } = useClipboardCopy(address);
+
+  const getAddress = useCallback(() => {
+    const addressElements = splitAddress(address);
+    if (full && !(addressElements instanceof Error)) {
+      return (
+        <MaskedAddress className={styles.address} address={address} full />
+      );
+    }
+    return <MaskedAddress className={styles.address} address={address} />;
```

Comment:
Why do we need to call `splitAddress` here if we're only checking for its error? The same seems to happen in MaskedAddress

Reply:
fair, I just copied over the old component and built upon it, I'll check if I can clean this up for a bit too